ci: create PR to udpate flake.lock every saturday

brianmay opened the pull request:
If this works it should create PRs that look a bit like https://github.com/brianmay/robotica-rust/pull/701

Caveats:

* Might have problems because I didn't create a personal token. Hoping this won't be required, but for personal project found it was. Particularly with dependabot. See https://github.com/DeterminateSystems/update-flake-lock?tab=readme-ov-file#with-a-personal-authentication-token
* Might have implications for running out of cache space, see #4297.
* Not sure what timezone this cron job uses.

brianmay commented:
There is a request for native nix flake support in dependabot, but don't hold your breath waiting for it to happen :-)

https://github.com/dependabot/dependabot-core/issues/7340

JakobLichterfeld commented on .github/workflows/update-flack-lock.yml:
```diff
@@ -0,0 +1,23 @@
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
```
Please use dep pinning: `actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2`

JakobLichterfeld commented on .github/workflows/update-flack-lock.yml:
```diff
@@ -0,0 +1,23 @@
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@main
```
What is the advantage of this nix installer? We use the cachix nix installer in our other nix workflow: https://github.com/teslamate-org/teslamate/blob/93cc34437c6466c993fde98bb271faaa9ed8a58d/.github/workflows/ensure_linting.yml#L16

brianmay replied:
Good question. I hadn't noticed that there are multiple competing solutions.

Perhaps I should change it to the one we already use.

I see a comparison at https://discourse.nixos.org/t/which-github-nix-installer-action-is-faster/25878

brianmay replied:
Supposedly https://github.com/nixbuild/nix-quick-install-action is fastest, it is single user only. But I think we would need to benchmark this before switching.

JakobLichterfeld replied:
Yeah, I found the above-mentioned benchmark after deciding for the cachix installer, which still claims to be fastest in most cases.

JakobLichterfeld commented on .github/workflows/update-flack-lock.yml:
```diff
@@ -0,0 +1,23 @@
+        uses: DeterminateSystems/nix-installer-action@main
+      - name: Update flake.lock
+        id: update
+        uses: DeterminateSystems/update-flake-lock@main
```
Please use dep pinning, `DeterminateSystems/update-flake-lock@v24`

JakobLichterfeld commented on .github/workflows/update-flack-lock.yml:
```diff
@@ -0,0 +1,23 @@
+  workflow_dispatch:
+  schedule:
+    # runs weekly on Saturday at 00:00
+    - cron: "0 0 * * 6"
```
This is UTC

JakobLichterfeld commented on .github/workflows/update-flack-lock.yml:
```diff
@@ -0,0 +1,23 @@
+on:
+  workflow_dispatch:
+  schedule:
+    # runs weekly on Saturday at 00:00
```
This comment does not reflect the cron job timer 😃

JakobLichterfeld reviewed (changes requested):
Thanks a lot! 
I do like the approach of having a Dependabot style for nix flake dependencies.

brianmay commented:
Thanks for the feedback. I agree with all points raised.

brianmay commented:
Hopefully I addressed all the feedback.

I left the `DeterminateSystems/nix-installer-action@main` open though so you can read what I said :-)

Obviously this has not been tested...

JakobLichterfeld reviewed (approved):
lgtm

JakobLichterfeld commented:
As feared, there are access problems: https://github.com/teslamate-org/teslamate/actions/runs/11813145746/job/32909622410#step:4:913

JakobLichterfeld commented:
Ok, we getting there:

https://github.com/teslamate-org/teslamate/actions/runs/11813338621/job/32910192341#step:4:921

JakobLichterfeld commented:
I changed our org settings: 
![image](https://github.com/user-attachments/assets/0bbcb099-d51e-4a8b-9abe-867162a321fb)

brianmay commented:
Great!
